src/main/scala/Constants.scala

```diff
@@ -21,5 +22,12 @@ object Instruction extends ChiselEnum {
 }
 
 object InstructionType extends ChiselEnum {
-  val IN_ERR, INST_R, INST_I, INST_S, INST_B, INST_U, INST_J, INST_Z = Value
+  val INST_I = Value((1 << 0).U)
+  val INST_S = Value((1 << 1).U)
+  val INST_B = Value((1 << 2).U)
+  val INST_U = Value((1 << 3).U)
+  val INST_J = Value((1 << 4).U)
+  val INST_Z = Value((1 << 5).U)
+  val INST_R = Value((1 << 6).U)
+  val IN_ERR = Value((1 << 7).U)
```

Comment:
From the RTL designer's review, your usage in decoder module needs a OH encoding which saves the extract comparing logic: (not (or (xor a b)))

Reply:
This is great... with https://github.com/chipsalliance/chisel3/pull/2261 it will be even better :)